scm-plugins/scm-git-plugin/src/main/js/RepositoryConfig.tsx

```diff
@@ -101,13 +101,22 @@ class RepositoryConfig extends React.Component<Props, State> {
       .get(configurationLink.href)
       .then(response => response.json())
       .then(payload =>
-        this.setState({
-          selectedBranchName: payload.defaultBranch,
-          nonFastForwardDisallowed: payload.nonFastForwardDisallowed,
-          disabled: !payload._links.update,
-          loadingDefaultBranch: false,
-          changed: false
-        })
+        this.setState(
+          {
+            selectedBranchName: payload.defaultBranch,
+            nonFastForwardDisallowed: payload.nonFastForwardDisallowed,
+            disabled: !payload._links.update,
+            loadingDefaultBranch: false,
+            changed: false
+          },
+          () => {
+            if (!this.state.selectedBranchName) {
+              this.setState({
+                selectedBranchName: this.state.branches?.filter((b: Branch) => b.defaultBranch)[0]?.name
+              });
+            }
+          }
+        )
```

Comment:
Wouldn't it be easier to make the check whether we have a default branch from the payload before we change the state? Maybe something like this:
```suggestion
      .then(payload => {
        const defaultBranch = payload.defaultBranch
          ? payload.defaultBranch
          : this.state.branches?.filter((b: Branch) => b.defaultBranch)[0]?.name;
        this.setState({
          selectedBranchName: defaultBranch,
          nonFastForwardDisallowed: payload.nonFastForwardDisallowed,
          disabled: !payload._links.update,
          loadingDefaultBranch: false,
          changed: false
        });
      })
```